Modules/arraymodule.c

```diff
@@ -197,20 +197,22 @@ b_setitem(arrayobject *ap, Py_ssize_t i, PyObject *v)
     /* PyArg_Parse's 'b' formatter is for an unsigned char, therefore
        must use the next size up that is signed ('h') and manually do
        the overflow checking */
-    if (!PyArg_Parse(v, "h;array item must be integer", &x))
+    if (!PyArg_Parse(v, "h:array('b').__setitem__", &x)) {
```

Comment:
ISTM that the message after the ';' might be set as an error message only in case the PyArg_* function is called with the wrong number of arguments, by code in vgetargs1_impl() (in Python/getargs.c).
but in our case, ISTM that the call to PyArg_Parse() can't reach that code in vgetargs1_impl().

anyway, IMHO, "array item must be integer" is not an appropriate error message for
`array.array('b').append(1 << 1000)`.
and I agree that `array('b').__setitem__()` isn't good.

maybe we should just add this?
```
if (PyErr_ExceptionMatches(PyExc_OverflowError)) {
    PyErr_SetString(PyExc_OverflowError,
                    "Python int does not fit in C char");
}
```